concept/type/impl/ThingTypeImpl.java

```diff
@@ -321,7 +330,13 @@ public void setPlays(RoleType roleType, RoleType overriddenType) {
     @Override
     public void unsetPlays(RoleType roleType) {
         final TypeEdge edge = vertex.outs().edge(Encoding.Edge.Type.PLAYS, ((RoleTypeImpl) roleType).vertex);
-        if (edge == null) return;
+        if (edge == null) {
+            if (this.getPlays().anyMatch(attr -> attr.equals(roleType)))
+                throw exception(GraknException.of(INVALID_UNDEFINE_INHERITED_PLAYS,
+                        this.getLabel().toString(), roleType.getLabel().toString()));
+            else throw exception(GraknException.of(INVALID_UNDEFINE_NONEXISTENT_PLAYS,
+                    this.getLabel().toString(), roleType.getLabel().toString()));
+        }
```

Comment:
Likewise, when undefining plays that aren't present, we find out which situation we are in and throw as appropriate.

Reply:
Same as above: formatting of brackets and line breaks